src/main/java/seedu/address/model/time/TimePeriod.java

```diff
@@ -94,6 +96,17 @@ public Day getSchoolDay() {
         return schoolDay;
     }
 
+    /**
+     * Splits the time period into a bunch of smaller hour blocks.
+     */
+    public List<HourBlock> fragmentIntoHourBlocks() {
```

Comment:
Interesting...

Reply:
Only `HourBlock`s contain `Commitment`s. As such, this is the only way to have a location with a time.

Particularly, `TimingRecommender` returns a list of longest timings as a list of `TimePeriod`s. We cannot attach locations to this. Hence this fragment method is born.